### ISSUE TITLE
Parallel build fails

### ISSUE THREAD

**Issue by SteffenL** [x2on/OpenSSL-for-iPhone#163]
macOS version: 10.13.3 (17D102)
Xcode version: 9.3 (9E145)

RAM: 32 GB
Disk: ~18 GB free space.

Build command:
```
./build-libssl.sh
```

Console output:
```
Build options
  OpenSSL version: 1.0.2l
  Architectures: x86_64 i386 arm64 armv7s armv7 tv_x86_64 tv_arm64
  iOS SDK: 11.3
  tvOS SDK: 11.3
  Number of make threads: 16
  Build location: /Users/sl/code/OpenSSL-for-iPhone

[snip]
Building openssl-1.0.2l for iPhoneSimulator 11.3 x86_64...
  Logfile: /Users/sl/code/OpenSSL-for-iPhone/bin/iPhoneSimulator11.3-x86_64.sdk/build-openssl-1.0.2l.log
  Configure...
  Patching Makefile...
  Make (using 16 thread(s))...
Building openssl-1.0.2l for iPhoneSimulator 11.3 i386...
  Logfile: /Users/sl/code/OpenSSL-for-iPhone/bin/iPhoneSimulator11.3-i386.sdk/build-openssl-1.0.2l.log
  Configure...
  Make (using 16 thread(s))...
Problem during make - Please check /Users/sl/code/OpenSSL-for-iPhone/bin/iPhoneSimulator11.3-i386.sdk/build-openssl-1.0.2l.log
```
Log: [build-openssl-1.0.2l.log](https://gist.githubusercontent.com/SteffenL/8bc59de685718003586fbd65f3affc98/raw/b89d826a22f718e2693542cba1e40388308b06cb/build-openssl-1.0.2l.log)

## Workaround

The build completes successfully as long as `--noparallel` is added to the command line. I can even build OpenSSL 1.0.2o without any problems.

```
./build-libssl.sh --version=1.0.2o --noparallel
```

**Comment by marcolabreu** [comment]
macOS 10.13.4 (17E199)
Xcode Version 9.3 (9E145)
RAM: 16 GB

Both commands bellow ran fine, although in a different macOS version, with half the threads.

```sh
> ./build-libssl.sh --version=1.0.2o

Build options
  OpenSSL version: 1.0.2o
  Architectures: x86_64 i386 arm64 armv7s armv7 tv_x86_64 tv_arm64
  iOS SDK: 11.3
  tvOS SDK: 11.3
  Number of make threads: 8
```

```sh
> ./build-libssl.sh --version=1.1.0h

Build options
  OpenSSL version: 1.1.0h
  Targets: ios-sim-cross-x86_64 ios-sim-cross-i386 ios64-cross-arm64 ios-cross-armv7s ios-cross-armv7 tvos-sim-cross-x86_64 tvos64-cross-arm64
  iOS SDK: 11.3
  tvOS SDK: 11.3
  Number of make threads: 8
  Configure options:  no-deprecated
```

**Comment by oneton** [comment]
After implementing the parallel build option, we observed several build issues on Travis when building 1.0.2. This is also the reason the --noparallel option was added. Usually trying again resolves the problem. I think the OpenSSL team also knew that there were some issues with a parallel build in 1.0.2 and they solved it in 1.1.0, so the 1.1.0 build should not run into issues with the parallel build.
